tt_metal/impl/kernels/kernel.cpp

```diff
@@ -295,6 +294,20 @@ bool Kernel::is_idle_eth() {
     return std::holds_alternative<EthernetConfig>(this->config()) && std::get<EthernetConfig>(this->config()).eth_mode == Eth::IDLE;
 }
 
+uint32_t Kernel::get_binary_packed_size(Device *device, int index) const {
+    // In testing situations we can query the size w/o a binary
+    return (this->binaries_.find(device->build_key()) != this->binaries_.end()) ?
+        this->binaries_.at(device->build_key())[index].get_packed_size() :
+        0;
```

Comment:
I'm not sure I fully understand what's going on here. Could I please get a quick explanation.

Reply:
binaries are stored per device, up to 3 (for math).  the if check is because some tests don't provide a program but set up, eg, CBs and then segfault above in validation.  I'd like us to clean this up w/ the program refactor and lazy state validation, I suspect it can go away (and the build key is going away too)